### PR TITLE
Fix UPCAddressofArraySubscriptGadget::getClaimedVarUseSites()

### PR DESCRIPTION
**Pull request by juanvazquez**
UPCAddressofArraySubscriptGadget::getClaimedVarUseSites should skip parentheses when accessing the DeclRefExpr, otherwise a crash happens with parenthesized references.

### CODE REVIEW

**Comment by juanvazquez** [comment]
Fixes #88405

**Review by jkorous-apple — APPROVED** [review]
LGTM
Thank you for fixing this!

**Comment by juanvazquez** [comment]
Ping

A gentle ping in case there is something missing on my side to get this merged. It's my first time contributing to LLVM. Thanks!

**Comment by ldionne** [comment]
I just approved the CI runs. I think this can be merged once the CI is all green. Whoever sees this first with push access can merge.

**Pull request merged**
